Test  for Product_property

Jeshan7 opened the pull request:
## Why?
     - Coverage for product_property_test.exs were low

## This change addresses the need by:
      - Updated the test cases in schema/product_property_test.exs
      - Wrote new test cases in schema/product_propert_test.exs 
           - Test cases for foreign_key constraint
           - Test cases for checking whether the property_id is unique or not 
           - Test cases for Create_changeset/2  with valid and invalid parameters 
           - Test cases Update_changeset/2 with valid and invalid parameters

[delivers #163215222]

## Checklist
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] I have read [CONTRIBUTING.md][contributing].
- [ ] My code follows the [style guidelines][contributing] of this project.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] My code does not generate any (new) [`credo`][credo] and compile-time warnings.
- [ ] I have updated the documentation wherever necessary.
- [ ] I have added tests to cover my changes.

[contributing]: https://github.com/aviabird/snitch/CONTRIBUTING.md
[credo]: https://github.com/rrrene/credo

Jeshan7 committed:
[#163215222]
      - Updated the test cases in schema/product_property_test.exs
      - Wrote new test cases in schema/product_propert_test.exs

SagarKarwande commented on apps/snitch_core/test/data/schema/product_property_test.exs:
```diff
@@ -1,56 +1,105 @@
-  end
+  describe "create_changeset/2" do
+    test "successfully with valid params", context do
+      %{params: params} = context
```
Don't pattern match the `params` separately. Patter match them in the test statement
```elixir
    test "successfully with valid params", %{params: params} do
```

SagarKarwande commented on apps/snitch_core/test/data/schema/product_property_test.exs:
```diff
@@ -1,56 +1,105 @@
+    end
+
+    test "fails for duplicate product_id", context do
+      %{params: params} = context
```
Pattern match the `params` in the test statement.

SagarKarwande commented on apps/snitch_core/test/data/schema/product_property_test.exs:
```diff
@@ -1,56 +1,105 @@
-    assert validity
-    assert {:ok, product_property} = Repo.insert(cset)
+    test "fails for non-existent property_id", context do
+      %{params: params} = context
```
Pattern match the `params` in the test statement

SagarKarwande commented on apps/snitch_core/test/data/schema/product_property_test.exs:
```diff
@@ -1,56 +1,105 @@
+        ]
+    end
+
+    test "successfully with valid params", %{changset: product_property, params: params} do
```
Here we should use already inserted Product Property for update changeset.
Also validate if the `update_changeset` has the changes we are about to update.

We are allowing to update the `property` and `value`. Both need to be tested.

Jeshan7 replied:
Ok

Jeshan7 replied:
In update_changeset/2 , params given to create_changeset is already inserted in the database

Jeshan7 commented on apps/snitch_core/test/data/schema/product_property_test.exs:
```diff
@@ -1,56 +1,105 @@
+          %ProductProperty{}
+          |> ProductProperty.create_changeset(params)
+          |> apply_changes(),
+
```
Params given to this changeset are already  present in db

SagarKarwande replied:
`apply_changes` does not do the DB operation, it only applies the changes from changeset to the Struct and returns the underlying struct.

[Doc - apply_changes/1](https://hexdocs.pm/ecto/Ecto.Changeset.html#apply_changes/1)

arjun289 commented on apps/snitch_core/test/data/schema/product_property_test.exs:
```diff
@@ -1,56 +1,84 @@
-    assert validity
-    assert {:ok, product_property} = Repo.insert(cset)
+    test "fails for non-existent product_id, property_id" do
+      params = %{
```
test statement shouldn't include `property_id`, it says the test is for non_existent product_id

Jeshan7 committed:
[#163215222]

Prior to this change,
  - params were patternmatched inside the test
  - In update_changset/2 params were given to empty struct

This change
  - params are  now patternmatched in test statement
  - In update_changeset/2 params are given to product_property
    struct which is already presnt in database

Jeshan7 committed:
[#163215222]

Prior to this change,
  Test statement
   - test "fails for non-existent product_id, property_id" do

This change
  Test statement
   -  test "fails for non-existent product_id" do
